### DISCUSSION TITLE
Compiling to .streamdeckplugin file?

### DISCUSSION THREAD

**Discussion by Annorax10** [elgatosf/elgato#73] [Help] [answered]
I've created a repository and manifest for a plugin that ONLY installs prebuilt profiles (there is no executable code in it whatsoever). I'm still very new to this (haven't coded anything since high school... in a year that started with 1), and I'm lost on this last step. How do I compile this into an installable plugin file?

**Comment by ChekTek** [comment]
Hi @Annorax10,

Do these profiles need to bundled together in plugin form? Typically, if there is no executable code, we'd recommend distributing them as profiles rather than a plugin. With the release of [Elgato Marketpace](https://marketplace.elgato.com/), we plan to soon support publishing profiles in addition to plugins, iconpacks, and other assets.

**Comment by Annorax10** [comment]
They need to be bundled, as when I have exported them for installation by other streamers who are on my shared channel, the profiles typically have things like Switch Profile actions changed to default profiles. I was looking at the plugin format as a way to install all profiles for this channel's menu and game element management system simultaneously without links between them getting messed up. I don't intend to make this publicly available, as it wouldn't be of value to anyone other than my stream team, and includes things like OBS scene collections that are referred to by OBS actions in the Streamdeck profiles.

**Comment by ChekTek** [comment, marked as answer]
I would probably still recommend just distributing them as 3 separate Stream Deck profiles and zipping them up in a folder letting the user install them in 3 steps, rather than bundling them as a plugin, as an empty plugin could have some unexpected results for Stream Deck. 

You can take a look at our [plugin template](https://github.com/elgatosf/streamdeck-plugin-template). The steps there should be able to get you started. You can then bundle it into a plugin using the [DistributionTool](https://docs.elgato.com/sdk/plugins/packaging).
